src/pool/srv_util.c

```diff
@@ -682,109 +675,128 @@ check_pool_targets(uuid_t pool_id, int *tgt_ids, int tgt_cnt, d_rank_t *pl_rank)
 		case PO_COMP_ST_DOWN:
 			nr_down++;
 			break;
+		case PO_COMP_ST_UPIN:
+			nr_upin++;
+			break;
+		case PO_COMP_ST_UP:
+			nr_up++;
+			break;
 		default:
 			break;
 		}
```

Comment:
Should this include the DRAIN state or NEW state also? DRAIN is similar to UPIN in that it's healthy/available... Not sure exactly what this function is doing... I don't think a target in the DRAIN state would be excluded or reintegrated though - that'd be the job of rebuild to finish performing the drain, then set directly to DOWNOUT. 

Same goes for NEW - it's essentially the same as UP, but can't be excluded directly - it has to either complete addition and move to UPIN or fail and be removed entirely.

Reply:
This reint callback is called when admin uses dmg command to replace a faulty device with a good device,  here we check the affected pool targets state (their original state should be DOWNOUT), if all pool targets are in either UP or UPIN, we consider the reintegration is already being successfully triggered.

NEW state is irrelevant in this case. For DWON/DRAIN/DOWNOUT, it's possible that some pool targets transit to DOWN/DRAIN/DOWNOUT again after it being reintegrated, but this is allowed, since BIO won't repeatedly trigger reintegration in such case.

Reply:
Sounds good.